packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2834,3 +2882,282 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// Signature for the callback that reports when the user changes the selection
+/// under a [SelectionListener].
+typedef SelectionListenerSelectionChangedCallback = void Function(List<SelectedContentRange<Object>>? selections);
+
+/// A [SelectionContainer] that allows the user to listen to selection changes
+/// for the child subtree it wraps under a [SelectionArea] or [SelectableRegion].
+///
+/// This widget should have an ancestor [SelectionArea] or [SelectableRegion]
+/// to be able to listen to selection changes in this widgets subtree.
+///
+/// This widget does not listen to selection changes of nested [SelectionArea]s
+/// or [SelectableRegion]s in its subtree because those widgets create their own
+/// [SelectionRegistrar]s that do not register to any ancestor registrar.
+///
+/// {@tool dartpad}
+/// This example shows how to color red the active selection
+/// under a [SelectionArea] or [SelectableRegion].
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.1.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to replace the active selection
+/// under a [SelectionArea] or [SelectableRegion] with a widget.
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.2.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [SelectionArea], which provides an overview of the selection system.
+///   * [SelectableRegion], which provides an overview of the selection system.
+class SelectionListener extends StatefulWidget {
+  /// Create a new [SelectionListener] widget.
+  const SelectionListener({
+    super.key,
+    required this.onSelectionChanged,
+    required this.child,
+  });
+
+  /// Called when the user changes the selection of children selectables
+  /// registered to its local [SelectionRegistrar].
+  final SelectionListenerSelectionChangedCallback onSelectionChanged;
+
+  /// The child widget this selection listener applies to.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  @override
+  State<SelectionListener> createState() => _SelectionListenerState();
+}
+
+class _SelectionListenerState extends State<SelectionListener> {
+  late final _SelectionListenerDelegate _selectionDelegate;
```

Comment:
Interesting I did not know this would work with `late`. When I remove `late` it complains about undefined `widget`. Is it preferable to initialize it inline instead of `initState`?

Reply:
Talked offline, prefer initializing inline instead of initState for readability.